### ISSUE TITLE
Provide functions to set outputs and environment variables

### ISSUE THREAD

**Issue by brechtm** [jannekem/run-python-script-action#3]
Just an idea that would make run-python-script-action even better.

I think the main use case is to set output or environment variables. run-python-script-action could provide functions to make this easier. You can prepend `import gha` to the script (perhaps controlled by an option) which provides these functions:

```python
def set_output(name, value):
    print(f'::set-output {name}=stable::{value}')

def set_env(name, value):
    with open(os.getenv('GITHUB_ENV'), 'a') as env:
        print(f'{name}={value}', file=env)
```

So then a user could do something like:

```yaml
- name: Find path to wheel
  uses: jannekem/run-python-script-action@v1
  with:
    script: |
      import os
      from pathlib import Path
      wheel_path = str(next(Path('dist').glob('*.whl')))
      gha.set_env('WHEEL_PATH', wheel_path)
```

Even better would be to also collect all output and environment variables and (optionally) print their names/values to the log, which can help with debugging.

**Comment by jannekem** [comment]
Sounds like a good idea. I might be able to have a look at this on the weekend. 👍

**Comment by jannekem** [comment]
I added a bunch of helper utility functions in PR #4. Would you like to test that they work for you too before I merge the changes?

**Comment by brechtm** [comment]
Wow, you really went to town with this suggestion! I didn't know about most of the GitHub Actions features exposed by the functions. I'll have to see if I can put them to good use in my workflow. I also like how the functions are available by default but can be disabled, if needed.

I adjusted my workflow to use the functions. That only tests `set_ouput`, `set_env`, `get_input` and `error`, but those work well so far.

Even though `get_input` seems to work well, [GitHub warns about _Unexpected inputs_](https://github.com/brechtm/rinohtype/runs/1801321207?check_suite_focus=true). So I will probably revert to getting the environment variables or make use of `${{ substitutions }}`.

**Comment by jannekem** [comment]
The [octokit/request-action](https://github.com/octokit/request-action/) has [a similar issue](https://github.com/octokit/request-action/issues/26). At some point GitHub made a change where all inputs that are not listed in the `action.yml` definition will cause that warning. It seems there is no way to suppress the warning when an action accepts arbitrary inputs.

**Comment by jannekem** [comment]
This is now released as `v1.2`, and the `v1` tag has also been moved to point to the latest version.
